tools/integration_tests/util/operations/file_operations.go

```diff
@@ -642,3 +632,63 @@ func SizeOfFile(filepath string) (size int64, err error) {
 
 	return (*fstat).Size(), nil
 }
+
+// Creates a temporary file (name-collision-safe) in /tmp with given content.
+// If gzipCompress is true, output file is a gzip-compressed file.
+// Caller is responsible for deleting the created file when done using it.
+// Failure cases:
+// 1. os.CreateTemp() returned error or nil handle
+// 2. gzip.NewWriter() returned nil handle
+// 3. Failed to write the content to the created temp file
+func CreateLocalTempFile(content string, gzipCompress bool) (string, error) {
+	contentSize := len(content)
+
+	// create appropriate name template for temp file
+	filenameTemplate := "testfile-*.txt"
+	if gzipCompress {
+		filenameTemplate += ".gz"
+	}
+
+	// create a temp file
+	f, err := os.CreateTemp(TmpDirectory, filenameTemplate)
+	if err != nil {
+		return "", fmt.Errorf("failed to create tempfile for template %s: %w", filenameTemplate, err)
+	}
+	if f == nil {
+		return "", fmt.Errorf("nil file handle returned from os.CreateTemp")
+	}
+	defer CloseFile(f)
+	filepath := f.Name()
+
+	if gzipCompress {
+		w := gzip.NewWriter(f)
+		if w == nil {
+			return "", fmt.Errorf("failed to create gzip writer for file %s", filepath)
+		}
+		defer func() {
+			if err := w.Close(); err != nil {
+				log.Printf("failed to close gzip writer for file %s: %v", filepath, err)
+			}
+		}()
+
+		// write the content created above as gzip
+		n, err := w.Write([]byte(content))
+		if err != nil {
+			return "", fmt.Errorf("failed to write content to %s using gzip-writer: %w", filepath, err)
+		}
+		if n != contentSize {
+			return "", fmt.Errorf("failed to write to gzip file %s. Content-size: %d bytes, wrote = %d bytes", filepath, contentSize, n)
+		}
+	} else {
+		// write the content created above as text
+		n, err := f.WriteString(content)
+		if err != nil {
+			return "", err
+		}
+		if n != contentSize {
+			return "", fmt.Errorf("failed to write to text file %s. Content-size: %d bytes, wrote = %d bytes", filepath, contentSize, n)
+		}
+	}
+
+	return filepath, nil
+}
```

Comment:
Reduce nesting: https://testing.googleblog.com/2017/06/code-health-reduce-nesting-reduce.html

Reply:
Would it be better to implement methods writeGzipToFile and writeTextToFile
 and make code something like this?

 if gzipCompress {
		return writeGzipToFile(f, filepath, content, contentSize)
	}

	return writeTextToFile(f, filepath, content, contentSize)
	
or it is good without these methods too?

Reply:
That'd be great!